build/management/commands/build_consensus_sequences.py

```diff
@@ -108,11 +108,17 @@ def main_func(self, positions, iteration,count,lock):
 
         if self.input_slug:
             families = ProteinFamily.objects.filter(slug__startswith=self.input_slug)
-        
+
+
         while count.value<len(families):
             with lock:
-                family = families[count.value]
-                count.value +=1 
+                # Check if the count changed before having obtained the lock
+                if count.value < len(families):
+                    family = families[count.value]
+                    count.value +=1
+                else:
+                    continue
```

Comment:
Note: this could alternatively be a break preventing another check of the while loop, but thought the code read nicer if it was a continue